node/pkg/logscribe/app.go

```diff
@@ -87,3 +131,106 @@ func bulkCopyLogs(ctx context.Context, logsChannel <-chan *[]api.LogInsertModel)
 		}
 	}
 }
+
+func (a *App) processLogs(ctx context.Context) {
+	ticker := time.NewTicker(a.processLogsInterval)
+
+	for {
+		select {
+		case <-ctx.Done():
+			ticker.Stop()
+			return
+		case <-ticker.C:
+			processedLogs := make([]LogInsertModelWithID, 0)
+			logMap := make(map[string]bool, 0)
+			totalLogs := 0
+			for {
+				logs, err := fetchLogs(ctx)
+				if err != nil || len(logs) == 0 {
+					break
+				}
+				totalLogs += len(logs)
+
+				for _, log := range logs {
+					hash := hashLog(log)
+					if !logMap[hash] {
+						processedLogs = append(processedLogs, log)
+						logMap[hash] = true
+					}
+				}
+
+				err = deleteLogs(ctx)
+				if err != nil {
+					break
+				}
+			}
+
+			log.Debug().Msgf("Processed %d logs with %d unique logs", totalLogs, len(processedLogs))
+			if len(processedLogs) > 0 {
+				a.createGithubIssue(ctx, processedLogs)
+			}
+		}
+	}
+}
+
+func fetchLogs(ctx context.Context) ([]LogInsertModelWithID, error) {
+	logs, err := db.QueryRows[LogInsertModelWithID](ctx, api.ReadLogs, map[string]any{
+		"limit": dbReadBatchSize,
+	})
+	if err != nil {
+		log.Error().Err(err).Msg("Failed to read logs")
+		return nil, err
+	}
+	return logs, nil
+}
+
+func deleteLogs(ctx context.Context) error {
+	err := retrier.Retry(func() error {
+		return db.QueryWithoutResult(ctx, api.DeleteLogs, map[string]any{
+			"limit": dbReadBatchSize,
+		})
+	}, 3, 1*time.Second, 3*time.Second)
+
+	if err != nil {
+		log.Error().Err(err).Msg("Failed to delete logs")
+		return err
+	}
+	return nil
+}
+
+func hashLog(log LogInsertModelWithID) string {
+	hash := sha256.New()
+	hash.Write([]byte(log.Service))
+	hash.Write([]byte(fmt.Sprintf("%d", log.Level)))
+	hash.Write([]byte(log.Message))
+	hash.Write(log.Fields)
+	return hex.EncodeToString(hash.Sum(nil))
+}
+
+func (a *App) createGithubIssue(ctx context.Context, logs []LogInsertModelWithID) {
+	if a.githubOwner == "test" {
+		return
+	}
+
+	var logsJson string
+	for _, entry := range logs {
+		entryJson, err := json.Marshal(entry)
+		if err != nil {
+			log.Error().Err(err).Msg("Failed to marshal log")
+			return
+		}
+		logsJson += string(entryJson) + "\n"
+	}
+
+	formattedBody := "```go\n" + logsJson + "\n```"
+	issueRequest := &github.IssueRequest{
+		Title:  github.String("Issues from Logscribe"),
+		Body:   github.String(formattedBody),
+		Labels: &[]string{"bug"},
+	}
+
+	_, resp, err := a.githubClient.Issues.Create(ctx, a.githubOwner, a.githubRepo, issueRequest)
+	if err != nil || resp.StatusCode != http.StatusCreated {
+		log.Error().Err(err).Msg("Failed to create github issue")
+	}
+}
```

Comment:
it'd be nice if github issue can be created per error. that's why I mentioned to count frequency or  number of errors. so we can prioritize certain errors to be made into issue

github issue title, body, label should also be customized based on log detail

Reply:
sounds good then I'll create separate issue for `error` level logs and a single issue for all `warn` level logs?

Reply:
good idea

Reply:
we don't need to create issues for every error logs, but based on the count rank or frequency rank, only top 5 or top 10 of them can be made into issue

Reply:
it'd be nice if it could be classified based on service such as api, delegator, node, etc

Reply:
thinking back again, lets ignore issue generation for warning for now